corgi/tasks/brew.py

```diff
@@ -469,17 +482,17 @@ def save_module(softwarebuild, build_data) -> ComponentNode:
     See CORGI-200, and CORGI-163"""
     meta_attr = build_data["meta"]["meta_attr"]
     obj, created = Component.objects.update_or_create(
-        name=build_data["meta"]["name"],
         type=build_data["type"],
-        arch=build_data["meta"].get("arch", ""),
+        name=build_data["meta"]["name"],
         version=build_data["meta"].get("version", ""),
         release=build_data["meta"].get("release", ""),
+        arch=build_data["meta"].get("arch", "noarch"),
         defaults={
-            "license_declared_raw": build_data["meta"].get("license", ""),
             "description": build_data["meta"].get("description", ""),
-            "software_build": softwarebuild,
+            "license_declared_raw": build_data["meta"].get("license", ""),
             "meta_attr": meta_attr,
-            "namespace": build_data["namespace"],
+            "namespace": Component.Namespace.REDHAT,
```

Comment:
I think it's safe to assume all container_image components are in the REDHAT namespace.

Reply:
What I was thinking was to adjust the Syft collector to set the arch correctly for detected RPM, instead of always setting it to arch="". Then it can to get_or_create() with the correct arch value. It looks like [Syft is capable](https://github.com/anchore/syft/tree/0774ad15e26bc9230dbfc393b3e7d6bd36b93cf7/syft/pkg/cataloger/rpm) of detecting RPM arch. I just neglected to pass it through from the collector in Corgi.